lib/adapters/http_server.js

```diff
@@ -103,21 +137,25 @@ var Read = source.extend({
 
         this.trigger('error', err);
         res.end();
-    },
-
-    teardown: function() {
-        if (this.server) {
-            this.server.close();
-        }
+    }
 
-        this.emit_eof();
+    teardown() {
+        return new Promise((resolve, reject) => {
+            if (this.server) {
+                this.server.close(() => {
```

Comment:
`this.server.close(resolve)` is the obnoxiously pithy way to express this, if you want.